### PR TITLE
Added Prime Factors of N in C

### PR DESCRIPTION
**Pull request by YashovardhanSiramdas**
**Changes:**
Added a program which prints all prime factors of any given Natural Number in C.

### CODE REVIEW

**Comment by YashovardhanSiramdas** [comment]
@InfiniteCoder Thanks for the suggestions !! :) I've renamed the variable "n" to "number". I'm deliberately declaring looping variables outside the loop as @arnavb mentioned earlier (#3137 ) that old C compilers will throw up a compilation error if looping variables are declared within the loop.

File: code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    for (i = 3; i <= sqrt(number); i = i + 2) {
+        while (number % i == 0) {
+            printf("%d ", i);
+            number = number / i;
+        }
+    }
+
+    if (number > 2)
+        printf ("%d ", number);
```
Comment (arnavb) [inline code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c]: I think returning a value would be better than printing one.

Comment (YashovardhanSiramdas) [reply code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c]: The variable "number" is getting modified inside the function multiple times. Printing prime factors is happening at 3 places (line 9,15,21)inside the function.So, I think it is not possible to return the value to main function at line 21. Moreover, the code is more readable if a single function prints all prime factors.

Comment (YashovardhanSiramdas) [reply code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c]: @arnavb What do you suggest about it? Is this program fine?

Comment (arnavb) [reply code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c]: I meant returning an array of prime numbers should be better. A function should only have 1 job.

**Comment by YashovardhanSiramdas** [comment]
@arnavb Now the function stores prime factors in array. Does this look good?

**Comment by YashovardhanSiramdas** [comment]
@arnavb I made suggested changes. Function will now return an array pointer.

**Comment by arnavb** [comment]
Good job!

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**